### PR TITLE
Notification fixes for closed app

### PR DESCRIPTION
**Pull request by paulwsmith**
https://snapyr.atlassian.net/browse/MEKA-615

## Notification helpers can boot their own Snapyr instance 

tl;dr fixes crash/non-displayed notifications when a Snapyr notification arrives on device after app has closed, in certain situations including use in React Native apps.

Scenario: a notification arrives, or is tapped, and the notification helper code no longer has access to the Snapyr instance.

This can happen if Snapyr is not instantiated early enough in application lifecycle to be available by the time the notification service or notification listener activity need to use it. (this is the case for React Native, where we can't initialize Snapyr until the main activity starts up)

To account for this, the main Snapyr init code now stashes minimal configs necessary to be able to communicate with the back end - namely, the write key and environment.

Then, in notification context, if there is not a valid Snapyr instance, it will use a helper to instantiate a one-off, minimal "helper" Snapyr instance. This disables things like lifecycle listeners, and uses its own one-off memory queue, so that notification tracking data can be sent to Snapyr without affecting anything about the main Snapyr instance that may be booted up soon afterward.

### Minor other fixes
* Improves timing of "replay" activity tracking; skips activity lifecycle tracking for internal Snapyr activity as this is only meant for tracking interaction, then launching the actual activity.
* Dev/debug fixes - remove batch upload debug log duplicate output; fix font to suppress font errors running sample app; fix stage engine endpoint

### CODE REVIEW

File: snapyr/src/main/java/com/snapyr/sdk/Snapyr.java [hunk 1]
```diff
@@ -214,8 +214,8 @@ public void handleMessage(Message msg) {
             this.sendQueue =
                     new BatchUploadQueue(application, 999999, 999, new BatchQueue.MemoryQueue());
         } else {
-        this.sendQueue =
-                new BatchUploadQueue(application, flushIntervalInMillis, flushQueueSize, null);
+            this.sendQueue =
```
Comment (onemorebsmith) [inline snapyr/src/main/java/com/snapyr/sdk/Snapyr.java:217]: Favorite change of the year

Comment (paulwsmith) [reply snapyr/src/main/java/com/snapyr/sdk/Snapyr.java:217]: Thank you... spotless worked hard to tweak all the spaces
